appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/LookupV1.java

```diff
@@ -0,0 +1,178 @@
+                throw (NamingException) ex;
+            }
+            throw wrap("Lookup failed for `" + jndiName + "` [" + mediaType + "]. Remote exception: ", ex);
+        }
+    }
+
+    static Exception deserialise(ErrorResponse error) {
+        Exception ex = null;
+        try {
+            ex = (Exception) Class.forName(error.exceptionType).getConstructor(String.class).newInstance(error.message);
+        } catch (Exception e) {
+            ex = new NamingException("Remote exception: " + error.exceptionType + ": " + error.message);
+        }
+        if (error.cause != null) {
+            ex.initCause(deserialise(error.cause));
+        }
+        return ex;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <C> C newProxy(Class<C> remoteBusinessInterface, WebTarget invoke, String mediaType, String jndiName,
+            Map<String, Object> jndiOptions) {
+        return (C) newProxyInstance(doPrivileged((PrivilegedAction<ClassLoader>) remoteBusinessInterface::getClassLoader),
+                new Class[] { remoteBusinessInterface },
+                new RemoteEjbOverHttpInvocationHandler(mediaType, invoke, jndiName, jndiOptions));
+    }
+
+    static final class RemoteEjbOverHttpInvocationHandler implements InvocationHandler {
+
+        private final String mediaType;
+        private final WebTarget invoke;
+        private final String jndiName;
+        private final Map<String, Object> jndiOptions;
+
+        public RemoteEjbOverHttpInvocationHandler(String mediaType, WebTarget invoke, String jndiName, Map<String, Object> jndiOptions) {
+            this.jndiName = jndiName;
+            this.jndiOptions = jndiOptions;
+            this.invoke = invoke;
+            this.mediaType = mediaType;
+        }
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            // Check for methods we should not proxy first
+            if (args == null && method.getName().equals("toString")) {
+                return toString();
+            }
+            if (args == null && method.getName().equals("hashCode")) {
+                // unique instance in the JVM, and no need to override
+                return hashCode();
+            }
+            if (args != null && args.length == 1 && method.getName().equals("equals")) {
+                // unique instance in the JVM, and no need to override
+                return equals(args[0]);
+            }
+            return invokeRemote(method, args);
+        }
+
+        private Object invokeRemote(Method method, Object[] args) throws Exception {
+            InvokeMethodRequest request = createRequest(method, args);
+            try (Response response = invoke.request(mediaType).buildPost(Entity.entity(request, mediaType)).invoke()) {
+                if (response.getStatus() == Status.OK.getStatusCode()) {
+                    return response.readEntity(InvokeMethodResponse.class).result;
```

Comment:
This will not he that easy, unfortunately. As noted in comment to JSON-B deserializer for this class, it wouldn't work for container generic return types (`List<Pojo>`). 
Using that deserializer (and making use of the provided return type) makes only sense if method's return type is an abstract class that is not a `Map` or `Collection`.